Promact.Oauth.Server/src/Promact.Oauth.Server/Repository/OAuthRepository/OAuthRepository.cs

```diff
@@ -40,24 +41,32 @@ private void Add(OAuth model)
             _oAuthDataRepository.Save();
         }
 
+
         /// <summary>
         /// To get details of a OAuth Access for an email and corresponding to app
         /// </summary>
         /// <param name="email"></param>
         /// <param name="clientId"></param>
         /// <returns></returns>
-        private OAuth GetDetails(string email,string clientId)
+        private OAuth GetDetails(string email, string clientId)
         {
-            var oAuth = _oAuthDataRepository.FirstOrDefault(x => x.userEmail == email && x.ClientId == clientId);
+            OAuth oAuth = _oAuthDataRepository.FirstOrDefault(x => x.userEmail == email && x.ClientId == clientId);
             return oAuth;
         }
 
-        private async Task<OAuth> OAuthClientChecking(string email,string clientId)
+
+        /// <summary>
+        /// Checks whether with this app email is register or not if  not new OAuth will be created.
+        /// </summary>
+        /// <param name="email"></param>
+        /// <param name="clientId"></param>
+        /// <returns></returns>
+        private async Task<OAuth> OAuthClientChecking(string email, string clientId)
         {
             //checking whether with this app email is register or not if  not new OAuth will be created.
-            var oAuth = GetDetails(email, clientId);
-            var app =  await _appRepository.GetAppDetails(clientId);
-            if (oAuth == null && app!=null)
+            OAuth oAuth = GetDetails(email, clientId);
```

Comment:
GetDetails is a private method and it just gets data from database. Do we need to give async support for such methods too ?